backend/src/routes/api/users/search.ts

```diff
@@ -0,0 +1,39 @@
+import express, { Request, Response, NextFunction } from "express";
+import {
+	auth,
+	getTokenFromReq,
+	getFromUserTokenAndAddIfNotFound,
+} from "../../../firebase";
+import { User } from "../../../models/User";
+
+const router = express.Router();
+
+router.get(
+	"/search",
+	async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            const searchString = req.query.searchString;
+
+            // check if the search string exists
+            if (!searchString) {
+                res.status(400).json({ message: "Search Paramater Not Provided" });
+                return;
+            }
+
+            // get the list of users that match the search string
+            const users = await User.find({
+                username: { $regex: searchString, $options: "i" },
+            });
+
+            // return the list of users
+            res.json(users);

```

Comment:
You dont want to return the whole users json thats a security vulnerability, we don't want to let everyone know what groupchats the user is in. Instead filter it out where you only return the users username, full name, id, and bio.

Reply:
also might be worth making a new function inside the Users.ts schema that returns the skeleton version of User

Reply:
Oh I thought we weren't storing any sensitive data on MongoDB since passwords and anything else would be on Firebase. Also a lot of the time UserID is used as a primary key access to lists of groupchats and not the other way around so I assumed that wasn't going to be stored per user.